File: Sort/BubbleSort.vb
Function sort(names As String(), size As Integer)
'Dim swap as boolean for do while Loop'
        Dim swap As Boolean
        swap = False
        Do
        'sets swap as false so we can tell if items still need to be sorted'
            swap = False
            'sorts through entire size of array'
            For i = 0 To size -1
            'compares names to sort them in order'
                If names(i) > names(i + 1) Then
                'performs a swap of the two strings'
                    Dim tmp As String
                    tmp = names(i)
                    names(i) = names(i + 1)
                    names(i + 1) = tmp
                    swap = True
                End If
            Next
            'checks if a swap had occurred and either continues or exits'
        Loop While (swap)
    End Function

End Module

File: Sort/Ruby/bubble_sort.rb
class BubbleSort

   def self.sort(array)
    n = array.length
    loop do
      swapped = false

      (n-1).times do |i|
        if array[i] > array[i+1]
          array[i], array[i+1] = array[i+1], array[i]
          swapped = true
        end
      end

      break if not swapped
    end

    array
  end
end

arr = [12,15,1,2,4,5,6,25,13,78,50,4,16,13,17]
print BubbleSort.sort(arr)
